### ISSUE TITLE
ACPI problems after --update (v2.1.0)

### ISSUE THREAD

**Issue by Bobbm** [theracermaster/Gigabyte-GA-Z77X-DSDT-Patch#73]
**System**
macOS 10.12.6
GA-Z77X.sh v2.1.0
GA-Z77X-UD3H
Intel Core i7-3.40
Intel HD Graphics 4000, no card

**Issue Type**
Errors during/after the post-install script --update

**Issue Description**
After (re-) running GA-Z77X.sh --update (v2.1.0)  boot chokes after two kextd wait cycles on AppleACPICPU, followed by a brief flash of an IOConsoleUsers message then a locked blank screen.

(Note that two days ago I posted here a separate Issues Report on other issues after first attempt at running --update. Main issue then was a config.plist with lots of weird redundancies which preventing booting. I eventually managed to reboot from a USB, and re-ran the update with GA-Z77X.sh. The resulting config.plist look normal so I marked that Issue as closed, and started new report here.)

**Comment by al3xtjames** [comment]
Can you post your EFI folder? Remove your serial number/etc from config.plist.

**Comment by Bobbm** [comment]
Thanks.

Since cannot boot, unless I am mistaken I believe the only way I can get at the EFI partition is to pull the HDD and attach it to another machine. Its midnight here, so that little job will be for tmrw morning.

**Comment by Bobbm** [comment]
note that at completion of this second attempt at running the script  --update, I got an error message: ".... AtherosE2200Ethernet.kext was installed improperly and cannot be used…". 

I got this same error on previous attempt, but I doubt my present ACPI panic is related to this kext install error. One issue at a time, I suspect that'll be easy to resolve once I can boot again.

**Comment by al3xtjames** [comment]
Hmm, not sure why you're seeing that. AtherosE2200Ethernet.kext shouldn't be installed to /Library/Extensions, it should be installed to the ESP and injected by Clover.

**Comment by Bobbm** [comment]
yes, i thought so too - in your approach all add-on kexts are injected by clover. This system was a clean install of sierra a few months ago using your script, but after install I use Apple migration assistant from a time machine B/U of an earlier install where I had wrstled a bit with ethernet. I though I had migrated only apps and data, but maybe somehow some junk from my old S/L/E got in there and was somehow noticed and error-flagged after running your script this time. no worries, once we resolve the ACPI boot issue that should be easy to sort out. will get a copy of EFI folder to you tmrw

**Comment by Bobbm** [comment]
Zipped copy of EFI folder (redacted) from SSD Sierra boot drive (which currently panics on AppleACPICPU)
[EFI-SSD.zip](https://github.com/al3xtjames/Gigabyte-GA-Z77X-macOS-Install/files/1202208/EFI-SSD.zip)

I used a USB linux boot to get this. If the problem is simple, perhaps I can just edit something in this EFI, copy it back into SSD EFI partition, and get 10.12.6 booting again. Possible?

**Comment by Bobbm** [comment]
here is my update script log
[update.txt.zip](https://github.com/al3xtjames/Gigabyte-GA-Z77X-macOS-Install/files/1202815/update.txt.zip)

**Comment by Bobbm** [comment]
instead of trying to get existing install's EFI fixed, I have been trying to do a clean install sierra with this script v2.1.0.  cannot get installer to boot. 

final message on verbose boot is waiting for an IOResourceMatch on boot-uuid-media, followed by USBMSC Identifier (non-unique) followed by a number. I

**Comment by Bobbm** [comment]
some progress here. I eventually got the current version of your installer to boot. Not completely sure what i did to resolve the most recent non-unique USBMSC Identifier issue from post above.  

I booted a linux system from a USB, and used efibootmgr to clean out most of the items in the boot list (thinking that possibly some confused items in the the NVRAM were somehow involved. I am not sure how UUIDs get assigned between board UEFI and Clover EFI boot.) 

I also unplugged everything USB except kbd/mouse, and removed all SATA devices except the target install SSD, and tried the install USB stick and kbd/mouse in several different USB busses. Eventually the UUID problem went away and the installer booted and worked. Copied the installer Clover folder to the EFI partition on the main SSD EFI partition, and the Sierra install now boots, though I haven't yet checked throughly what else works.

Will try the GA-Z77X.sh V2.1.0 script again tonight (this time making sure to keep a copy of the current EFI folder!) Will report here again if my earlier ACPI panic (first post, above) re-occurs.

**Comment by Bobbm** [comment]
success! V2.1.0 script with --install option ran fine, boots fine now from SSD. Ran script after after clean install (install was followed by migration of data, apps, user settings from time machine BU)
